src/pip/_internal/configuration.py

```diff
@@ -210,8 +210,15 @@ def save(self) -> None:
             # Ensure directory exists.
             ensure_dir(os.path.dirname(fname))
 
-            with open(fname, "w") as f:
-                parser.write(f)
+            # Ensure directory's permission(need to be writeable)
+            try:
+                with open(fname, "w") as f:
+                    parser.write(f)
+            except IOError as error:
```

Comment:
Do we also want to catch OSError here?

Reply:
IOError is just an alias of `OSError`(at least in recent Python versions) so IMO catching `OSError` is more accurate.

Reply:
I’ll just change it then